cmd/mdatagen/README.md

```diff
@@ -55,6 +55,27 @@ Below are some more examples that can be used for reference:
 
 You can run `cd cmd/mdatagen && $(GOCMD) install .` to install the `mdatagen` tool in `GOBIN` and then run `mdatagen metadata.yaml` to generate documentation for a specific component or you can run `make generate` to generate documentation for all components.
 
+### Generate a package with an alternate name
+
+By default, `mdatagen` will generate a package called `metadata` in the `internal` directory. If you want to generate a package with a different name, you can use the `--package_name` flag to provide an alternate name.
+
+```go
+//go:generate mdatagen --package_name custom custom.yaml
```

Comment:
Why not have it in the yaml config?

Reply:
Good idea, I'm happy to change it to that instead.

Reply:
Let's try to keep the entire config in the file and avoid mixing flags and file if possible.

Reply:
Done in [d3f5e4e](https://github.com/open-telemetry/opentelemetry-collector/pull/11232/commits/d3f5e4eddc6dd5f02f108876d92f7e7544238c5f)